docs/operator-manual/user-management/okta.md

```diff
@@ -134,17 +134,33 @@ data:
     1. Choose the matching options you need, one example is:
         * e.g. to match groups starting with `argocd-` you'd return an `ID Token` using your scope name from step 3 (e.g. `groups`) where the groups name `matches` the `regex` `argocd-.*`
     ![Groups Claim](../../assets/groups-claim.png)
-1. Edit the `argocd-cm` and configure the `data.oidc.config` section:
+1. Create a new Okta OIDC Web Application
+    1. Enable refresh token grant type
+    1. Edit the sign-in redirect-uri to `https://external.path.to.argocd.io/auth/callback` (this can also point to localhost if you are testing)
+    1. Edit the sign-out redirect-uri to `https://external.path.to.argocd.io`
+    ![Okta web app configuration](../../assets/okta-webapp-configuration.png)
+    1. Save
+    1. After creation, go to the "Sign On" tab and select "OpenID Connect ID Token"
+    1. Edit the groups claim filter to "Starts with: *". This will pass all groups in the JWT. You may customize this to the groups that are useful to your RBAC use case.
+    ![Okta OIDC token settings](../../assets/okta-oidc-token-settings.png)
+1. Edit the `argocd-cm` and configure the following sections:
 
 <!-- markdownlint-disable MD046 -->
 ```yaml
-oidc.config: |
-  name: Okta
-  issuer: https://yourorganization.oktapreview.com
-  clientID: 0oaltaqg3oAIf2NOa0h3
-  clientSecret: ZXF_CfUc-rtwNfzFecGquzdeJ_MxM4sGc8pDT2Tg6t
-  requestedScopes: ["openid", "profile", "email", "groups"]
-  requestedIDTokenClaims: {"groups": {"essential": true}}
+config :
+  # The URL that your users reach ArgoCD on. Can be https://localhost:8080 if you are testing
+  url: https://external.path.to.argocd.io
+  oidc.config: |
+    name: Okta
+    issuer: https://yourorganization.oktapreview.com # or https://yourorganization.okta.com
+    clientID: replace-this
+    clientSecret: replace-this
```

Comment:
Can you provide examples that point to external secrets? I think it's best to encourage that rather than letting folks assume they should hard-code secrets.

Reply:
You're right! I've updated the instructions